src/mui/input/SelectManyInput.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component, PropTypes } from 'react';
+import ChipInput from 'material-ui-chip-input';
+import FieldTitle from '../../util/FieldTitle';
+
+/**
+ * An Input component for a array
+ *
+ * @example
+ * <SelectManyInput source="first_name" />
+ *
+ * Pass possible options as an array of objects in the 'choices' attribute.
+ *
+ * By default, the options are built from:
+ *  - the 'id' property as the option value,
+ *  - the 'name' property an the option text
+ * @example
+ * const choices = [
+ *    { id: 'M', name: 'Male' },
+ *    { id: 'F', name: 'Female' },
+ * ];
+ * <SelectManyInput source="gender" choices={choices} />
+ *
+ * You can also customize the properties to use for the option name and value,
+ * thanks to the 'optionText' and 'optionValue' attributes.
+ * @example
+ * const choices = [
+ *    { _id: 123, full_name: 'Leo Tolstoi', sex: 'M' },
+ *    { _id: 456, full_name: 'Jane Austen', sex: 'F' },
+ * ];
+ * <SelectManyInput source="author_id" choices={choices} optionText="full_name" optionValue="_id" />
+ *
+ * The object passed as `options` props is passed to the material-ui-chip-input component
+ * @see https://github.com/TeamWertarbyte/material-ui-chip-input
+ */
+export class SelectManyInput extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            values: this.resolveValues(props.input.value || []),
+        };
+    }
+
+    handleBlur = () => {
+        var extracted = this.extractIds(this.state.values);
+        this.props.onBlur(extracted);
+        this.props.input.onBlur(extracted);
+    };
+    handleFocus = () => {
+        var extracted = this.extractIds(this.state.values);
+        this.props.onFocus(extracted);
+        this.props.input.onFocus(extracted);
+    };
+
+    handleAdd = (newValue) => {
+        this.state.values = [...this.state.values, newValue];
+        this.handleChange(this.state.values);
+    };
+
+    handleDelete = (newValue) => {
+        this.state.values = this.state.values.filter(v => (v[this.props.optionValue] !== newValue));
+        this.handleChange(this.state.values);
+    };
+
+    handleChange = (eventOrValue) => {
+        var extracted = this.extractIds(eventOrValue);
+        this.props.onChange(extracted);
```

Comment:
what's this `onChange` for? Same for the other custom handlers (`onBlur`, etc): they don't exist in other components, so it doesn't make sense to add them just for that component